src/syntax-tree/ast.ts

```diff
@@ -858,6 +858,69 @@ export class ElseStatement extends Statement {
     }
 }
 
+export class ImportStatement extends Statement {
+    private moduleNameIndex: number = -1;
+    private itemNameIndex: number = -1;
+    constructor(moduleName: string = "   ", itemName: string = "   ") {
+        super();
+
+        this.tokens.push(new NonEditableTkn("from ", this, this.tokens.length));
+        this.moduleNameIndex = this.tokens.length;
+        this.tokens.push(new EditableTextTkn(moduleName, /[a-zA-Z]/g, this, this.tokens.length));
+        this.tokens.push(new NonEditableTkn(" import ", this, this.tokens.length));
+        this.itemNameIndex = this.tokens.length;
+        this.tokens.push(new EditableTextTkn(itemName, /[a-zA-Z]/g, this, this.tokens.length));
+
+        this.subscribe(
+            CallbackType.onFocusOff,
+            new Callback(() => {
+                this.onFocusOff({ module: this.getModule() });
+            })
+        );
+
+        this.subscribe(
+            CallbackType.delete,
+            new Callback(() => {
+                this.onDelete(this.getModule()); //TODO: I don't like this either, but we need the module there
+            })
+        );
+    }
+
+    validateContext(validator: Validator, providedContext: Context): InsertionType {
+        //TODO: This is the same for most Statements so should probably move this functionality into the parent class
+        return validator.onEmptyLine(providedContext) ? InsertionType.Valid : InsertionType.Invalid;
+    }
+
+    getImportModuleName(): string {
+        return this.tokens[this.moduleNameIndex].getRenderText();
+    }
+    getImportItemName(): string {
+        return this.tokens[this.itemNameIndex].getRenderText();
+    }
+
+    onFocusOff(args: any): void {
+        if (this.getImportModuleName() !== "" && this.getImportItemName() !== "") {
+            //TODO: Not efficient, but the only way to improve this is to constantly maintain an updated "imported" status
+            //on the construct requiring an import, which is tedious so I left it for now. If this ever becomes an issue, that is the solution.
+            args.module.validator.validateImports();
+        }
+    }
+
+    setImportModule(txt: string) {
+        (this.tokens[this.moduleNameIndex] as EditableTextTkn).setEditedText(txt);
+    }
+
+    setImportItem(txt: string) {
+        (this.tokens[this.itemNameIndex] as EditableTextTkn).setEditedText(txt);
+    }
+
+    private onDelete(module: Module) {
```

Comment:
Needs to become a shared functionality coming from some parent class. VarAssignment needs this too and more constructs will require it in the future so it would be good to enforce this requirement through inheritance and come up with a universal way of handling it in the parent since the patterns of usage are the same.